src/services/counters/WeakCounter.ts

```diff
@@ -0,0 +1,33 @@
+import {AbstractCounter} from "./AbstractCounter";
+export class WeakCounter extends AbstractCounter {
+
+  constructor(protected name: string, protected storage: string,
+              protected initialValue: number, protected currentValue: number,
+              protected concurrency: number, protected dmr: any) {
+    super(name, storage, initialValue, currentValue, dmr);
+  }
+
+  isStrong(): boolean {
+    return false;
+  }
+
+  hasBounds(): boolean {
+    return false;
+  }
+
+  isWeak(): boolean {
+    return true;
+  }
+
+  getConcurrency(): number {
+    return this.concurrency;
+  }
+
+  getLowerBound(): number {
+    return undefined;
+  }
+
+  getUpperBound(): number {
+    return undefined;
```

Comment:
Ok, this is more of a philosophical difference in opinion, no? I like dealing with ICounter interface rather than check types, it simplifies view code...

Reply:
Well if you are going to go down inheritance route to reduce code repetition then the code how it currently is doesn't really make sense. However, if you would prefer just to have implementations of `ICounter.ts` and have some unnecessary code duplication I can live with that, you make the call. But, it doesn't make sense to have a strange hybrid where we use some inheritance but duplicate most of the code anyway. 

Furthermore, you actually only call `hasBounds` in one view, so you're writing a lot of boilerplate code to avoid `instanceof StrongCounter`. With regards to `isStrong()` the only places you use it in the view can be simplified by adding a `toString` method to `IContainer` and providing STRONG/WEAK as the return value in the appropriate impl.